### PR TITLE
Refactor asset formular 

### PR DESCRIPTION
**Pull request by yixingjia**

### CODE REVIEW

File: aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java [hunk 1]
```diff
@@ -426,38 +439,50 @@ public void aggregateServerPDU() {
...
-                        isNeedUpdated = true;
+               Map<String, Map<String, String>> pduFormulasMap = metricsFormulaToMap(pduFormulaInfo);
+               if(pduFormulasMap != null) {
+                  if(pduIds.size() != pduFormulasMap.keySet().size()) {
+                     pduFormulasMap = generatePduformulaForServer(pduIds);
+                     isNeedUpdated = true;
+                  }else {
+                     for(String pduAssetId : pduIds) {
+                        if(pduFormulasMap.get(pduAssetId) == null) {
+                           pduFormulasMap = generatePduformulaForServer(pduIds);
```
Comment (yixingjia) [inline aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java:473]: ?

File: aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java [hunk 2]
```diff
@@ -746,4 +779,24 @@ public String getPositionInfo(Asset asset) {
       return positionInfo.toString();
    }
 
+   private String metricsFormulatoString(Map<String, Map<String, String>> pduFormulasMap) {
+      String pduFormulaInfo = null;
```
Comment (yixingjia) [inline aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java:783]: its a common method,not pud specify.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 3]
```diff
@@ -749,4 +755,24 @@ private String getSensorPositionInfo(Asset asset) {
       }
       return positionInfo.toString();
    }
+
+   private Map<String, Map<String, String>> metricsFormulaToMap(String pduFormulasInfo){
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:759]: refactor

File: nlyte-worker/src/main/java/com/vmware/flowgate/nlyteworker/scheduler/job/NlyteDataService.java [hunk 4]
```diff
@@ -326,15 +329,31 @@ private void removeInActiveData(FacilitySoftwareConfig nlyte) {
...
             }
-          }
-         server.setMetricsformulars(formulars);
+         }
+
+         try {
+            formulars.put(FlowgateConstant.PDU, mapper.writeValueAsString(pduFormulars));
+         } catch (JsonProcessingException e) {
+            logger.error("Format metric formula error", e);
+         }
```
Comment (yixingjia) [inline nlyte-worker/src/main/java/com/vmware/flowgate/nlyteworker/scheduler/job/NlyteDataService.java:338]: refactor

File: poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java [hunk 5]
```diff
@@ -327,14 +327,29 @@ public void removeSensorMetaData(FacilitySoftwareConfig powerIQ) {
                }
             }
          }
-         server.setMetricsformulars(formulars);
+         try {
+            formulars.put(FlowgateConstant.SENSOR, mapper.writeValueAsString(sensorFormulars));
+         } catch (JsonProcessingException e) {
+            logger.error("Format metric formula error" ,e);
+         }
+         server.setMetricsformulas(formulars);
```
Comment (yixingjia) [inline poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java:335]: refactor

File: poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java [hunk 6]
```diff
@@ -649,22 +664,30 @@ public void saveSensorAssetsToFlowgate(Map<String, Asset> exsitingSensorAssets,
...
             Map<String, Map<String, String>> sensorFormulars = generateNewMetricformular(sensorAsset, positionInfo);
             if(sensorFormulars.isEmpty()) {
                continue;
             }
-            formulars.put(FlowgateConstant.SENSOR, sensorFormulars);
+            try {
+               formulars.put(FlowgateConstant.SENSOR, mapper.writeValueAsString(sensorFormulars));
+            } catch (JsonProcessingException e) {
+               logger.error("Format metric formula error" ,e);
+            }
```
Comment (yixingjia) [inline poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java:678]: refactor

File: vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java [hunk 7]
```diff
@@ -386,34 +387,37 @@ public void queryHostMetrics(SDDCSoftwareConfig vc) {
...
-            asset.getMetricsformulars();
-      Map<String, Map<String, String>> metrics =
-            metricsFormulars.get(FlowgateConstant.HOST_METRICS);
+      Map<String, String> metricsFormulars = asset.getMetricsformulas();
+      Map<String, String> metrics = null;
+      try {
+         metrics = mapper.readValue(metricsFormulars.get(FlowgateConstant.HOST_METRICS), new TypeReference<Map<String, String>>() {});
+      } catch (JsonProcessingException e) {
+         logger.error("Format host info map error", e);
+      }
```
Comment (yixingjia) [inline vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java:396]: refactor

File: vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java [hunk 8]
```diff
@@ -386,34 +387,37 @@ public void queryHostMetrics(SDDCSoftwareConfig vc) {
...
+
+         // storage
+         metrics.put(MetricName.SERVER_STORAGEIORATEUSAGE, assetId);
+         metrics.put(MetricName.SERVER_STORAGEUSAGE, assetId);
+         metrics.put(MetricName.SERVER_STORAGEUSED, assetId);
+         try {
+            metricsFormulars.put(FlowgateConstant.HOST_METRICS, mapper.writeValueAsString(metrics));
+         } catch (JsonProcessingException e) {
+            logger.error("Format host info map error", e);
+         }
```
Comment (yixingjia) [inline vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java:419]: refactor

File: aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java (hunk 1 above)
Comment (Pengpengwanga) [reply aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java:475]: Add break.
Done.

**Pull request merged**